src/epcpm/staticmodbusmodel.py

```diff
@@ -0,0 +1,1356 @@
+
+    type_uuid = epyqlib.attrsmodel.attr_uuid(
+        default=None,
+        allow_none=True,
+        human_name="Type",
+        data_display=epyqlib.attrsmodel.name_from_uuid,
+        list_selection_root="staticmodbus types",
+    )
+
+    uuid = epyqlib.attrsmodel.attr_uuid()
+
+    def __attrs_post_init__(self):
+        super().__init__()
+
+    def can_drop_on(self, node):
+        return isinstance(node, epyqlib.pm.parametermodel.Parameter)
+
+    def child_from(self, node):
+        self.parameter_uuid = node.uuid
+
+        return None
+
+    can_delete = epyqlib.attrsmodel.childless_can_delete
+    all_addable_types = epyqlib.attrsmodel.empty_all_addable_types
+    addable_types = epyqlib.attrsmodel.empty_addable_types
+    remove_old_on_drop = epyqlib.attrsmodel.default_remove_old_on_drop
+    internal_move = epyqlib.attrsmodel.default_internal_move
+    check = epyqlib.attrsmodel.check_just_children
+
+
+@graham.schemify(tag="data_point_bitfield", register=True)
+@epyqlib.attrsmodel.ify()
+@epyqlib.utils.qt.pyqtify()
+@attr.s(hash=False)
+class DataPointBitfield(epyqlib.treenode.TreeNode):
+    parameter_uuid = create_parameter_uuid_attribute()
+
+    children = attr.ib(
+        factory=list,
+        metadata=graham.create_metadata(
+            field=graham.fields.MixedList(
+                fields=(
+                    marshmallow.fields.Nested(graham.schema(DataPointBitfieldMember)),
+                )
+            ),
+        ),
+    )
+
+    # TODO: though this only really makes sense as one of the bitfield types
+    type_uuid = epyqlib.attrsmodel.attr_uuid(
+        default=None,
+        allow_none=True,
+        human_name="Type",
+        data_display=epyqlib.attrsmodel.name_from_uuid,
+        list_selection_root="staticmodbus types",
+    )
+
+    # block_offset = attr.ib(
+    #     default=0,
+    #     converter=int,
+    #     metadata=graham.create_metadata(
+    #         field=marshmallow.fields.Integer(),
+    #     ),
+    # )
+
+    size = create_size_attribute()
+
+    uuid = epyqlib.attrsmodel.attr_uuid()
+
+    def __attrs_post_init__(self):
+        super().__init__()
+
+    def can_drop_on(self, node):
+        return isinstance(
+            node,
+            (
+                DataPointBitfieldMember,
+                epyqlib.pm.parametermodel.Parameter,
+            ),
+        )
+
+    def can_delete(self, node=None):
+        if node is None:
+            return self.tree_parent.can_delete(node=self)
+
+        return True
+
+    def child_from(self, node):
+        if isinstance(node, epyqlib.pm.parametermodel.Parameter):
+            self.parameter_uuid = node.uuid
+            return None
+
+        return node
+
+    remove_old_on_drop = epyqlib.attrsmodel.default_remove_old_on_drop
+    internal_move = epyqlib.attrsmodel.default_internal_move
+    check = epyqlib.attrsmodel.check_just_children
+
+
+def check_block_offsets_and_length(self):
```

Comment:
I'm not sure what form this will have in the new stuff.

Reply:
Commented out in 8acdd06f64bf270239f398f132d1924ae155b029